06_Comparing_Object_and_Truth_Tables.ipynb

```diff
@@ -159,124 +161,88 @@
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "6decf8ca",
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "# Get positions, PSF magnitudes and fluxes, cModel magnitudes and fluxes,\n",
-    "#   and some flags from the Object table:\n",
-    "results_obj = service.search(\"SELECT objectId, ra, dec, mag_g, mag_r,\\\n",
-    "                              mag_i, mag_g_cModel, mag_r_cModel, mag_i_cModel,\\\n",
-    "                              psFlux_g, psFlux_r, psFlux_i, cModelFlux_g,\\\n",
-    "                              cModelFlux_r, cModelFlux_i tract, patch,\\\n",
-    "                              extendedness, good, clean\\\n",
-    "                              FROM dp01_dc2_catalogs.object\\\n",
-    "                              WHERE CONTAINS(POINT('ICRS', ra, dec),\\\n",
-    "                              CIRCLE('ICRS', 62.0, -37.0, 0.1)) = 1 \")\n",
-    "# results_tab = results.to_table()\n",
-    "# results_tab  # To print the table to the screen."
+    "For those who are not familiar with databases, the simplest way of accomplishing the task might seem to be querying the two tables (Object and Truth-Match) separately, then matching them afterwards. This would work, but it is not the best use of the resources that are available. As seen in the [Advanced TAP/ADQL Usage in the Portal Aspect](https://dp0-1.lsst.io/tutorials-examples/index-portal-advanced.html#examples-dp0-1-portal-advanced) tutorial, the table JOIN can be done directly with ADQL. Not only will this save you a few steps, it should also execute much faster.\n",
+    "\n",
+    "For this exploration, we will select a small region of sky around a random RA, Dec position. The following cell reads data centered on (RA, Dec) = (62.0, -37.0) degrees, within a radius of 0.1 degrees, from both the Object and Truth-Match table, then joins database entries where `match_objectId` from Truth-Match equals `objectId` from the Object table. Note that we are selecting only a subset of the columns seen in the schema above. You can add or remove columns as you wish.\n",
+    "\n",
+    "Note that for the Object table we select all objects within the cone-shaped region of interest. In the Truth-Match table, we restrict the results to objects satisfying \"match_objectId >= 0 AND is_good_match = 1\". According to the Truth-Match schema above, the `is_good_match` flag is \"True if this object--truth matching pair satisfies all matching criteria\" as laid out in the [DESC DC2 Release Note](https://ui.adsabs.harvard.edu/abs/2021arXiv210104855L/abstract). We'll use that to select \"good\" matches. In the column description for `match_objectId` from above, it says \"objectId of the matching object entry (-1 for unmatched truth entries).\" Thus the criterion \"match_objectId >= 0\" removes the unmatched entries, leaving us with only the truth-table entries that were detected and appear in the Object table.\n"
```

Comment:
Check that this (and other above) line lengths do not violate the coding guidelines.

Reply:
Markdown cells aren't checked by the notebook versions of flake8. The only lines in this notebook that exceed the line length guidelines are in the last two plotting cells. But for readability, I think they are best left as they are.